packages/shared/src/components/sidebar/SidebarList.tsx

```diff
@@ -26,6 +28,18 @@ function SidebarList({
 }: SidebarListProps): ReactElement {
   const { onRequestClose } = useModalContext();
 
+  const sidebarGroups = items.reduce((acc, item) => {
+    const group = item.group ?? defaultGroup;
+
+    if (!acc[group]) {
+      acc[group] = [];
+    }
+
+    acc[group].push(item);
+
+    return acc;
+  }, {} as Record<string, SidebarListItemProps[]>);
```

Comment:
Added support for grouping sidebar list items, due to requested changes in mobile design. By default if no groups are provided the list is rendered flat as before (that is what `Default` group is for).